{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "2b1bc2fe",
    "outputId": "9432c469-3767-412f-a84e-e586c51ceda3"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Mounted at /content/drive\n"
     ]
    }
   ],
   "source": [
    "#Login to the user's google driver account\n",
    "from google.colab import drive\n",
    "drive.mount('/content/drive')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "B0VVo3silxnw",
    "outputId": "419844b5-97b1-49b5-eb9c-3f74df22feff"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: apt-key is deprecated. Manage keyring files in trusted.gpg.d instead (see apt-key(8)).\n",
      "Executing: /tmp/apt-key-gpghome.0COGMktcfG/gpg.1.sh --keyserver keyserver.ubuntu.com --recv-keys E298A3A825C0D65DFD57CBB651716619E084DAB9\n",
      "gpg: key 51716619E084DAB9: \"Michael Rutter <marutter@gmail.com>\" 1 new signature\n",
      "gpg: Total number processed: 1\n",
      "gpg:         new signatures: 1\n",
      "Repository: 'deb https://cloud.r-project.org/bin/linux/ubuntu focal-cran40/'\n",
      "Description:\n",
      "Archive for codename: focal-cran40/ components: \n",
      "More info: https://cloud.r-project.org/bin/linux/ubuntu\n",
      "Adding repository.\n",
      "Press [ENTER] to continue or Ctrl-c to cancel.\n",
      "Adding deb entry to /etc/apt/sources.list.d/archive_uri-https_cloud_r-project_org_bin_linux_ubuntu-jammy.list\n",
      "Adding disabled deb-src entry to /etc/apt/sources.list.d/archive_uri-https_cloud_r-project_org_bin_linux_ubuntu-jammy.list\n",
      "Get:1 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ InRelease [3,626 B]\n",
      "Get:2 https://cloud.r-project.org/bin/linux/ubuntu focal-cran40/ InRelease [3,622 B]\n",
      "Hit:3 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease\n",
      "Get:4 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ Packages [46.6 kB]\n",
      "Get:5 http://security.ubuntu.com/ubuntu jammy-security InRelease [110 kB]\n",
      "Get:6 https://cloud.r-project.org/bin/linux/ubuntu focal-cran40/ Packages [90.3 kB]\n",
      "Hit:7 http://archive.ubuntu.com/ubuntu jammy InRelease\n",
      "Get:8 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [119 kB]\n",
      "Get:9 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy InRelease [18.1 kB]\n",
      "Get:10 http://security.ubuntu.com/ubuntu jammy-security/main amd64 Packages [1,186 kB]\n",
      "Get:11 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [109 kB]\n",
      "Get:12 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages [1,455 kB]\n",
      "Hit:13 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy InRelease\n",
      "Get:14 http://security.ubuntu.com/ubuntu jammy-security/universe amd64 Packages [1,013 kB]\n",
      "Get:15 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages [1,279 kB]\n",
      "Hit:16 https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu jammy InRelease\n",
      "Get:17 http://archive.ubuntu.com/ubuntu jammy-updates/restricted amd64 Packages [1,419 kB]\n",
      "Get:18 http://archive.ubuntu.com/ubuntu jammy-updates/multiverse amd64 Packages [49.8 kB]\n",
      "Hit:19 https://ppa.launchpadcontent.net/ubuntugis/ppa/ubuntu jammy InRelease\n",
      "Get:20 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main Sources [2,231 kB]\n",
      "Get:21 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy/main amd64 Packages [1,145 kB]\n",
      "Fetched 10.3 MB in 9s (1,173 kB/s)\n",
      "Reading package lists... Done\n",
      "Hit:1 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ InRelease\n",
      "Hit:2 https://cloud.r-project.org/bin/linux/ubuntu focal-cran40/ InRelease\n",
      "Hit:3 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease\n",
      "Hit:4 http://security.ubuntu.com/ubuntu jammy-security InRelease\n",
      "Hit:5 http://archive.ubuntu.com/ubuntu jammy InRelease\n",
      "Hit:6 http://archive.ubuntu.com/ubuntu jammy-updates InRelease\n",
      "Hit:7 http://archive.ubuntu.com/ubuntu jammy-backports InRelease\n",
      "Hit:8 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy InRelease\n",
      "Hit:9 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy InRelease\n",
      "Hit:10 https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu jammy InRelease\n",
      "Hit:11 https://ppa.launchpadcontent.net/ubuntugis/ppa/ubuntu jammy InRelease\n",
      "Reading package lists... Done\n",
      "Reading package lists... Done\n",
      "Building dependency tree... Done\n",
      "Reading state information... Done\n",
      "The following additional packages will be installed:\n",
      "  r-base-core r-recommended\n",
      "Suggested packages:\n",
      "  elpa-ess r-doc-info | r-doc-pdf r-mathlib r-base-html\n",
      "Recommended packages:\n",
      "  r-base-html r-doc-html\n",
      "The following packages will be upgraded:\n",
      "  r-base r-base-core r-recommended\n",
      "3 upgraded, 0 newly installed, 0 to remove and 31 not upgraded.\n",
      "Need to get 27.2 MB of archives.\n",
      "After this operation, 57.3 kB of additional disk space will be used.\n",
      "Get:1 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ r-base-core 4.3.2-1.2204.0 [27.2 MB]\n",
      "Get:2 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ r-base 4.3.2-1.2204.0 [46.4 kB]\n",
      "Get:3 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ r-recommended 4.3.2-1.2204.0 [2,592 B]\n",
      "Fetched 27.2 MB in 2s (17.1 MB/s)\n",
      "debconf: unable to initialize frontend: Dialog\n",
      "debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 78, <> line 3.)\n",
      "debconf: falling back to frontend: Readline\n",
      "debconf: unable to initialize frontend: Readline\n",
      "debconf: (This frontend requires a controlling tty.)\n",
      "debconf: falling back to frontend: Teletype\n",
      "dpkg-preconfigure: unable to re-open stdin: \n",
      "(Reading database ... 120874 files and directories currently installed.)\n",
      "Preparing to unpack .../r-base-core_4.3.2-1.2204.0_amd64.deb ...\n",
      "Unpacking r-base-core (4.3.2-1.2204.0) over (4.3.1-4.2204.0) ...\n",
      "Preparing to unpack .../r-base_4.3.2-1.2204.0_all.deb ...\n",
      "Unpacking r-base (4.3.2-1.2204.0) over (4.3.1-4.2204.0) ...\n",
      "Preparing to unpack .../r-recommended_4.3.2-1.2204.0_all.deb ...\n",
      "Unpacking r-recommended (4.3.2-1.2204.0) over (4.3.1-4.2204.0) ...\n",
      "Setting up r-base-core (4.3.2-1.2204.0) ...\n",
      "Installing new version of config file /etc/R/Makeconf ...\n",
      "debconf: unable to initialize frontend: Dialog\n",
      "debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 78.)\n",
      "debconf: falling back to frontend: Readline\n",
      "Setting up r-recommended (4.3.2-1.2204.0) ...\n",
      "Setting up r-base (4.3.2-1.2204.0) ...\n",
      "Processing triggers for man-db (2.10.2-1) ...\n",
      "Processing triggers for hicolor-icon-theme (0.17-2) ...\n",
      "Reading package lists... Done\n",
      "Building dependency tree... Done\n",
      "Reading state information... Done\n",
      "The following NEW packages will be installed:\n",
      "  codonw\n",
      "0 upgraded, 1 newly installed, 0 to remove and 31 not upgraded.\n",
      "Need to get 140 kB of archives.\n",
      "After this operation, 430 kB of additional disk space will be used.\n",
      "Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 codonw amd64 1.4.4-6 [140 kB]\n",
      "Fetched 140 kB in 1s (119 kB/s)\n",
      "debconf: unable to initialize frontend: Dialog\n",
      "debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 78, <> line 1.)\n",
      "debconf: falling back to frontend: Readline\n",
      "debconf: unable to initialize frontend: Readline\n",
      "debconf: (This frontend requires a controlling tty.)\n",
      "debconf: falling back to frontend: Teletype\n",
      "dpkg-preconfigure: unable to re-open stdin: \n",
      "Selecting previously unselected package codonw.\n",
      "(Reading database ... 120874 files and directories currently installed.)\n",
      "Preparing to unpack .../codonw_1.4.4-6_amd64.deb ...\n",
      "Unpacking codonw (1.4.4-6) ...\n",
      "Setting up codonw (1.4.4-6) ...\n",
      "Processing triggers for man-db (2.10.2-1) ...\n",
      "Installing packages into ‘/usr/local/lib/R/site-library’\n",
      "(as ‘lib’ is unspecified)\n",
      "also installing the dependencies ‘pixmap’, ‘sp’, ‘Rcpp’, ‘RcppArmadillo’, ‘ade4’, ‘segmented’, ‘matrixStats’\n",
      "\n",
      "trying URL 'https://cran.rstudio.com/src/contrib/pixmap_0.4-12.tar.gz'\n",
      "Content type 'application/x-gzip' length 34637 bytes (33 KB)\n",
      "==================================================\n",
      "downloaded 33 KB\n",
      "\n",
      "trying URL 'https://cran.rstudio.com/src/contrib/sp_2.1-1.tar.gz'\n",
      "Content type 'application/x-gzip' length 1244612 bytes (1.2 MB)\n",
      "==================================================\n",
      "downloaded 1.2 MB\n",
      "\n",
      "trying URL 'https://cran.rstudio.com/src/contrib/Rcpp_1.0.11.tar.gz'\n",
      "Content type 'application/x-gzip' length 2994004 bytes (2.9 MB)\n",
      "==================================================\n",
      "downloaded 2.9 MB\n",
      "\n",
      "trying URL 'https://cran.rstudio.com/src/contrib/RcppArmadillo_0.12.6.6.0.tar.gz'\n",
      "Content type 'application/x-gzip' length 1403237 bytes (1.3 MB)\n",
      "==================================================\n",
      "downloaded 1.3 MB\n",
      "\n",
      "trying URL 'https://cran.rstudio.com/src/contrib/ade4_1.7-22.tar.gz'\n",
      "Content type 'application/x-gzip' length 3365194 bytes (3.2 MB)\n",
      "==================================================\n",
      "downloaded 3.2 MB\n",
      "\n",
      "trying URL 'https://cran.rstudio.com/src/contrib/segmented_1.6-4.tar.gz'\n",
      "Content type 'application/x-gzip' length 178319 bytes (174 KB)\n",
      "==================================================\n",
      "downloaded 174 KB\n",
      "\n",
      "trying URL 'https://cran.rstudio.com/src/contrib/matrixStats_1.0.0.tar.gz'\n",
      "Content type 'application/x-gzip' length 210653 bytes (205 KB)\n",
      "==================================================\n",
      "downloaded 205 KB\n",
      "\n",
      "trying URL 'https://cran.rstudio.com/src/contrib/protr_1.7-0.tar.gz'\n",
      "Content type 'application/x-gzip' length 1901580 bytes (1.8 MB)\n",
      "==================================================\n",
      "downloaded 1.8 MB\n",
      "\n",
      "trying URL 'https://cran.rstudio.com/src/contrib/rDNAse_1.1-1.tar.gz'\n",
      "Content type 'application/x-gzip' length 1290276 bytes (1.2 MB)\n",
      "==================================================\n",
      "downloaded 1.2 MB\n",
      "\n",
      "trying URL 'https://cran.rstudio.com/src/contrib/seqinr_4.2-30.tar.gz'\n",
      "Content type 'application/x-gzip' length 3538250 bytes (3.4 MB)\n",
      "==================================================\n",
      "downloaded 3.4 MB\n",
      "\n",
      "trying URL 'https://cran.rstudio.com/src/contrib/matrixTests_0.2.3.tar.gz'\n",
      "Content type 'application/x-gzip' length 79169 bytes (77 KB)\n",
      "==================================================\n",
      "downloaded 77 KB\n",
      "\n",
      "* installing *source* package ‘pixmap’ ...\n",
      "** package ‘pixmap’ successfully unpacked and MD5 sums checked\n",
      "** using staged installation\n",
      "** R\n",
      "** inst\n",
      "** byte-compile and prepare package for lazy loading\n",
      "** help\n",
      "*** installing help indices\n",
      "** building package indices\n",
      "** testing if installed package can be loaded from temporary location\n",
      "** testing if installed package can be loaded from final location\n",
      "** testing if installed package keeps a record of temporary installation path\n",
      "* DONE (pixmap)\n",
      "* installing *source* package ‘sp’ ...\n",
      "** package ‘sp’ successfully unpacked and MD5 sums checked\n",
      "** using staged installation\n",
      "** libs\n",
      "using C compiler: ‘gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0’\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-H0vbME/r-base-4.3.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c Rcentroid.c -o Rcentroid.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-H0vbME/r-base-4.3.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c gcdist.c -o gcdist.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-H0vbME/r-base-4.3.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c init.c -o init.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-H0vbME/r-base-4.3.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c pip.c -o pip.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-H0vbME/r-base-4.3.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c pip2.c -o pip2.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-H0vbME/r-base-4.3.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c sp_xports.c -o sp_xports.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-H0vbME/r-base-4.3.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c surfaceArea.c -o surfaceArea.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-H0vbME/r-base-4.3.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c zerodist.c -o zerodist.o\n",
      "gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -flto=auto -ffat-lto-objects -flto=auto -Wl,-z,relro -o sp.so Rcentroid.o gcdist.o init.o pip.o pip2.o sp_xports.o surfaceArea.o zerodist.o -L/usr/lib/R/lib -lR\n",
      "installing to /usr/local/lib/R/site-library/00LOCK-sp/00new/sp/libs\n",
      "** R\n",
      "** data\n",
      "** demo\n",
      "** inst\n",
      "** byte-compile and prepare package for lazy loading\n"
     ]
    }
   ],
   "source": [
    "#add the GPG key.\n",
    "# adding the secret key to the set of trusted keys\n",
    "!sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys E298A3A825C0D65DFD57CBB651716619E084DAB9\n",
    "#add repository\n",
    "# add r-project repository to apt repository\n",
    "!sudo add-apt-repository 'deb https://cloud.r-project.org/bin/linux/ubuntu focal-cran40/'\n",
    "# update apt repository\n",
    "!sudo apt-get update\n",
    "#install R\n",
    "!sudo apt install r-base\n",
    "#install codonw\n",
    "!sudo apt-get install codonw\n",
    "#installation of required packages\n",
    "!echo \"install.packages(c('protr','rDNAse','seqinr','matrixTests'))\" > installPackages.R\n",
    "!Rscript installPackages.R\n",
    "#install biopython\n",
    "!pip install biopython\n",
    "!pip install rdkit\n",
    "!pip install -q condacolab"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "coIi3dHT5ClV"
   },
   "outputs": [],
   "source": [
    "#necessary dependencies\n",
    "from Bio import SeqIO\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Seq import Seq\n",
    "import os\n",
    "from os import listdir\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import sys\n",
    "import re\n",
    "import csv\n",
    "import condacolab"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "hI8P4RcN0ayU"
   },
   "outputs": [],
   "source": [
    "condacolab.install()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "17zI8bkw0ot5"
   },
   "outputs": [],
   "source": [
    "#iFeature package\n",
    "!pip3 install iFeatureOmegaCLI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "ca5be3e6"
   },
   "outputs": [],
   "source": [
    "#organization code to process here\n",
    "organism_abbr = [\"dde\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "0facdd2a"
   },
   "source": [
    "# Section in charge of the recovery of sequences in Kegg"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 369
    },
    "id": "a8cc61a3",
    "outputId": "d0dd0d9b-0480-47ab-f394-a976cf936f85"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cloning into 'extract_feature_and_sequences'...\n",
      "remote: Enumerating objects: 283, done.\u001b[K\n",
      "remote: Counting objects: 100% (283/283), done.\u001b[K\n",
      "remote: Compressing objects: 100% (256/256), done.\u001b[K\n",
      "remote: Total 283 (delta 28), reused 275 (delta 25), pack-reused 0\u001b[K\n",
      "Receiving objects: 100% (283/283), 18.11 MiB | 28.02 MiB/s, done.\n",
      "Resolving deltas: 100% (28/28), done.\n",
      "/content/extract_feature_and_sequences\n"
     ]
    },
    {
     "ename": "NameError",
     "evalue": "name 'organism_abbr' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-1-3310c98c35c7>\u001b[0m in \u001b[0;36m<cell line: 6>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0mget_ipython\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrun_line_magic\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'cd'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'extract_feature_and_sequences/'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0;31m#execution of the sequence extraction script\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 6\u001b[0;31m \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mrange\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0morganism_abbr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      7\u001b[0m   \u001b[0mget_ipython\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msystem\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'bash extract_organism_sequence.sh {organism_abbr[i]}'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      8\u001b[0m \u001b[0;31m#Changing the current directory\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'organism_abbr' is not defined"
     ]
    }
   ],
   "source": [
    "#cloning of the gitlab repository which contains the script for retrieving the sequences\n",
    "!git clone https://gitlab.com/fofack/extract_feature_and_sequences.git\n",
    "#Changing the current directory\n",
    "%cd extract_feature_and_sequences/\n",
    "#execution of the sequence extraction script\n",
    "for i in range (len(organism_abbr)):\n",
    "  !bash extract_organism_sequence.sh {organism_abbr[i]}\n",
    "#Changing the current directory\n",
    "%cd ../.."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "7b1bed9d"
   },
   "source": [
    "# Section that contains the various functions useful for preprocessing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "4c4b72ef"
   },
   "outputs": [],
   "source": [
    "#function to retrieve the elements of a directory\n",
    "def find_dir( path_to_dir):\n",
    "    dirnames = listdir(path_to_dir)\n",
    "    return [ dirname for dirname in dirnames if dirname.endswith]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "44badd03"
   },
   "outputs": [],
   "source": [
    "#function to retrieve fasta files from a directory\n",
    "def find_fasta_filenames( path_to_dir, suffix=\".fasta\" ):\n",
    "    filenames = listdir(path_to_dir)\n",
    "    return [ filename for filename in filenames if filename.endswith( suffix ) ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "50f98c02"
   },
   "outputs": [],
   "source": [
    "#cleaning of gene sequences to remove sequences <=30 characters and descriptions in front of identifiers\n",
    "def parsing_sequence_dna(path_input, path_out):\n",
    "    with open(path_input) as handle:\n",
    "        with open(path_out, 'a') as clean_file:\n",
    "            for record in SeqIO.parse(handle, \"fasta\"):\n",
    "                if(len(record.seq) > 30):\n",
    "                    clean_file.write('>'+ \"%s\" %record.id[4:].upper()+'\\n')\n",
    "                    clean_file.write(\"%s\" %record.seq+'\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "7c793441"
   },
   "outputs": [],
   "source": [
    "#cleanup of protein sequences to remove sequences <=30 characters and descriptions in front of identifiers\n",
    "def parsing_sequence_protein(path_input, path_out):\n",
    "    with open(path_input) as handle:\n",
    "        with open(path_out, 'a') as clean_file:\n",
    "            for record in SeqIO.parse(handle, \"fasta\"):\n",
    "                if(len(record.seq) > 30):\n",
    "                    clean_file.write('>'+ \"%s\" %record.id[4:].upper()+'\\n')\n",
    "                    clean_file.write(\"%s\" %record.seq+'\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "20663b4a"
   },
   "outputs": [],
   "source": [
    "#cleaning of sequences with certain codons that do not correspond to any amino acid\n",
    "def delete_sequence_id(path_input, path_out,tab_id):\n",
    "    with open(path_input) as handle:\n",
    "        with open(path_out, 'a') as clean_file:\n",
    "            for record in SeqIO.parse(handle, \"fasta\"):\n",
    "                operator = record.id in tab_id\n",
    "                if(operator == False):\n",
    "                    clean_file.write('>'+ \"%s\" %record.id+'\\n')\n",
    "                    clean_file.write(\"%s\" %record.seq+'\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "23cbc0e3"
   },
   "outputs": [],
   "source": [
    "#fusion of the two functions of cleaning gene and protein sequences\n",
    "def main_parsing_sequences(path_dir_input,path_dir_out):\n",
    "    path_data = find_fasta_filenames(path_dir_input)\n",
    "    for i in range(len(path_data)):\n",
    "        if(path_data[i] == \"dna_sequence.fasta\"):\n",
    "            print(path_dir_out+path_data[i])\n",
    "            parsing_sequence_dna(path_dir_input+path_data[i],path_dir_out+\"/\"+\"dna_sequence.fasta\")\n",
    "        else:\n",
    "            parsing_sequence_protein(path_dir_input+path_data[i],path_dir_out+\"/\"+\"protein_sequence.fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "a5015f8f"
   },
   "outputs": [],
   "source": [
    "# function to retain only sequences that are in both genes and proteins\n",
    "def clean_sequences(gene_sequence, protein_sequence,path_out):\n",
    "    #load dataset\n",
    "    #aa\n",
    "    fastaSequences = SeqIO.parse(open(protein_sequence), 'fasta')\n",
    "    seqaaDict_E=dict()\n",
    "    for fasta in fastaSequences:\n",
    "        name, sequence = fasta.id, str(fasta.seq)\n",
    "        seqaaDict_E[name]=sequence\n",
    "    #print(seqaaDict_E)\n",
    "    #aa\n",
    "    fastaSequences = SeqIO.parse(open(gene_sequence), 'fasta')\n",
    "    seqntDict_E=dict()\n",
    "    for fasta in fastaSequences:\n",
    "        name, sequence = fasta.id, str(fasta.seq)\n",
    "        seqntDict_E[name]=sequence\n",
    "        #print(seqntDict_E)\n",
    "    index=0\n",
    "    fileSeqAA=open(path_out+\"EssentialsequenceAA-clean.fasta\", 'a')\n",
    "    fileSeqNT=open( path_out+\"EssentialsequenceNT-clean.fasta\", 'a')\n",
    "    for cle in seqntDict_E.keys():\n",
    "        if cle in seqaaDict_E:\n",
    "            #print(cle)\n",
    "            sequenceProt=seqaaDict_E[cle].upper()\n",
    "            sequenceGene=seqntDict_E[cle].upper()\n",
    "            gene_Locus=cle\n",
    "            recordAA=SeqRecord(Seq(sequenceProt), id=str(gene_Locus),description=\"\")\n",
    "            recordNT=SeqRecord(Seq(sequenceGene), id=str(gene_Locus),description=\"\")\n",
    "            SeqIO.write(recordAA, fileSeqAA, \"fasta\")\n",
    "            SeqIO.write(recordNT, fileSeqNT, \"fasta\")\n",
    "            index=index+1\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "dcc6abfb"
   },
   "outputs": [],
   "source": [
    "#function to transform an array of array into a simple array\n",
    "def transform_in_simple_array(tab):\n",
    "    probleme_codonw = []\n",
    "    p = re.compile(\"[A-Z]+[0-9]*[\\_]?[A-Z]*[0-9]+[\\_]?[0-9]*[\\-]?[A-Z]*[0-9]*\")\n",
    "    for i in range (len(tab)):\n",
    "        if(len(p.findall(tab[i]))):\n",
    "            probleme_codonw.append(p.findall(tab[i]))\n",
    "    id_probleme = np.array(probleme_codonw).flatten()\n",
    "    return id_probleme"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "6donxd6upPcn"
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "zHbZx-g5pP8p"
   },
   "source": [
    "##Feature Methode"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "n037Yjx11Nfs"
   },
   "outputs": [],
   "source": [
    "import iFeatureOmegaCLI as iF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "aLh6KrZopVvE"
   },
   "outputs": [],
   "source": [
    "def get_feature_dna(path_input_sequence, path_out):\n",
    "    dna_descriptors = [\"NAC\",\"ANF\",\"ENAC\",\"MMI\",\"LPDF\",\"EIIP\",\"DPCP\",\"TPCP\",\n",
    "                       \"Kmer type 1\",\"RCKmer type 1\",\"PseEIIP\",\"PseKNC\",\n",
    "                       \"PCPseDNC\",\"PCPseTNC\",\"SCPseDNC\",\"SCPseTNC\",\"PSTNPss\",\n",
    "                       \"PSTNPds\",\"CKSNAP type 1\", \"DAC\",\"DACC\", \"DCC\", \"EGAAC\",\n",
    "                       \"TAC\", \"TACC\", \"TCC\",]\n",
    "    dna = iF.iDNA(path_input_sequence)\n",
    "    for i in range (len(dna_descriptors)):\n",
    "        dna.get_descriptor(dna_descriptors[i])\n",
    "        if(dna_descriptors[i] == \"Kmer type 1\"):\n",
    "          protein.to_csv(path_out+\"EssentialFeature_collection_\"+\"Kmer1\"+\".csv\",index=True, header=True)\n",
    "        elif (dna_descriptors[i] == \"RCKmer type 1\"):\n",
    "          protein.to_csv(path_out+\"EssentialFeature_collection_\"+\"RCKmer1\"+\".csv\",index=True, header=True)\n",
    "        elif (dna_descriptors[i] == \"CKSNAP type 1\"):\n",
    "          protein.to_csv(path_out+\"EssentialFeature_collection_\"+\"CKSNAP\"+\".csv\",index=True, header=True)\n",
    "        else:\n",
    "          dna.to_csv(path_out+\"EssentialFeature_collection_\"+dna_descriptors[i]+\".csv\",index=True, header=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "H76IdRugptPv"
   },
   "outputs": [],
   "source": [
    "def get_feature_peptide(path_file_sequence, path_out):\n",
    "    protein_descriptors = [\"CTDC\",\"CTDD\",\"CTDT\",\"DDE\",\"EGAAC\",\"GAAC\",\"AC\",\"CC\",\n",
    "                           \"ACC\",\"CKSAAP type 1\",\"KSCTriad\",\"DistancePair\",\n",
    "                           \"CKSAAGP type 1\",\"GDPC type 1\",\"GTPC type 1\",\"ASDC\",\n",
    "                           \"DPC type 1\",\"DPC type 2\",\"ZScale\", \"AAIndex\",\n",
    "                           \"TPC type 1\",\"TPC type 2\", \"ASDC\"]\n",
    "    protein = iF.iProtein(path_file_sequence)\n",
    "    for i in range (len(protein_descriptors)):\n",
    "      protein.get_descriptor(protein_descriptors[i])\n",
    "      if(protein_descriptors[i] == \"CKSAAP type 1\"):\n",
    "        protein.to_csv(path_out+\"EssentialFeature_collection_\"+\"CKSAAP\"+\".csv\",index=True, header=True)\n",
    "      elif (protein_descriptors[i] == \"DPC type 1\"):\n",
    "        protein.to_csv(path_out+\"EssentialFeature_collection_\"+\"DPC1\"+\".csv\",index=True, header=True)\n",
    "      elif (protein_descriptors[i] == \"DPC type 2\"):\n",
    "        protein.to_csv(path_out+\"EssentialFeature_collection_\"+\"DPC2\"+\".csv\",index=True, header=True)\n",
    "      elif (protein_descriptors[i] == \"TPC type 1\"):\n",
    "        protein.to_csv(path_out+\"EssentialFeature_collection_\"+\"TPC1\"+\".csv\",index=True, header=True)\n",
    "      elif (protein_descriptors[i] == \"TPC type 2\"):\n",
    "        protein.to_csv(path_out+\"EssentialFeature_collection_\"+\"TPC2\"+\".csv\",index=True, header=True)\n",
    "      elif (protein_descriptors[i] == \"GDPC type 1\"):\n",
    "        protein.to_csv(path_out+\"EssentialFeature_collection_\"+\"GDPC1\"+\".csv\",index=True, header=True)\n",
    "      elif (protein_descriptors[i] == \"GTPC type 1\"):\n",
    "        protein.to_csv(path_out+\"EssentialFeature_collection_\"+\"GTPC1\"+\".csv\",index=True, header=True)\n",
    "      else:\n",
    "        protein.to_csv(path_out+\"EssentialFeature_collection_\"+protein_descriptors[i]+\".csv\",index=True, header=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "ly3uvVlxptg_"
   },
   "outputs": [],
   "source": [
    "def main_iFeature(input_dna, out_dna, input_peptide, out_peptide):\n",
    "    get_feature_dna(input_dna, out_dna)\n",
    "    get_feature_peptide(input_peptide, out_peptide)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "XrOX330xk8Pn"
   },
   "outputs": [],
   "source": [
    "def split_data(path_dir, path_out):\n",
    "    file = find_dir(path_dir)\n",
    "    #file.remove('.ipynb_checkpoints')\n",
    "    for i in range (len(file)):\n",
    "        print(file[i])\n",
    "        with open(path_dir+file[i],\"r\") as handle:\n",
    "            data = csv.reader(handle)\n",
    "            rows = [r for r in data]\n",
    "            rows[0].pop(0)\n",
    "            with open(path_out+file[i],\"w\") as f:\n",
    "                writer = csv.writer(f)\n",
    "                for j in range(len(rows)):\n",
    "                    writer.writerow(rows[j])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "93d8a34f"
   },
   "source": [
    "# Section responsible for feature extraction using R language packages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "q9DvLIOgRg6R"
   },
   "outputs": [],
   "source": [
    "def main(path_input, path_out):\n",
    "    organism = find_dir(path_input)\n",
    "    for i in range (len(organism)):\n",
    "        print(\"################\"+\" \"+\"Feature de l'organisme\"+\" \"+organism[i]+\" \"+\"################\")\n",
    "        os.mkdir(path_out+organism[i])\n",
    "        os.mkdir(\"/content/results\"+organism[i])\n",
    "        main_parsing_sequences(path_input+organism[i]+\"/\", path_out+organism[i]+\"/\")\n",
    "        clean_sequences(path_out+organism[i]+\"/\"+\"dna_sequence.fasta\", path_out+organism[i]+\"/\"+\"protein_sequence.fasta\", path_out+organism[i]+\"/\")\n",
    "        chemin = path_out+organism[i]\n",
    "        pb = !codonw {chemin}/EssentialsequenceNT-clean.fasta results/{organism[i]}/EssentialcodonwFeature.out results/{organism[i]}/EssentialcodonwFeature.blk -nomenu -nowarn -all_indices -human\n",
    "        id_probleme = transform_in_simple_array(pb)\n",
    "        print(id_probleme)\n",
    "        if(len(id_probleme) > 0):\n",
    "            !mkdir clean_sequence_copy\n",
    "            !mkdir clean_sequence_copy/{organism[i]}\n",
    "            !mv /content/sequences_clean/{organism[i]}/dna_sequence.fasta clean_sequence_copy/{organism[i]}/\n",
    "            !mv /content/sequences_clean/{organism[i]}/protein_sequence.fasta clean_sequence_copy/{organism[i]}/\n",
    "            !rm /content/sequences_clean/{organism[i]}/EssentialsequenceNT-clean.fasta\n",
    "            !rm /content/sequences_clean/{organism[i]}/EssentialsequenceAA-clean.fasta\n",
    "            !rm /content/results/{organism[i]}/EssentialcodonwFeature.out\n",
    "            !rm /content/results/{organism[i]}/EssentialcodonwFeature.blk\n",
    "            delete_sequence_id(\"/content/clean_sequence_copy/\"+organism[i]+\"/\"+\"dna_sequence.fasta\", path_out+organism[i]+\"/\"+\"dna_sequence.fasta\", id_probleme)\n",
    "            delete_sequence_id(\"/content/clean_sequence_copy/\"+organism[i]+\"/\"+\"protein_sequence.fasta\", path_out+organism[i]+\"/\"+\"protein_sequence.fasta\", id_probleme)\n",
    "            clean_sequences(path_out+organism[i]+\"/\"+\"DNA_sequence.fasta\", path_out+organism[i]+\"/\"+\"protein_sequence.fasta\", path_out+organism[i]+\"/\")\n",
    "            chemin = path_out+organism[i]\n",
    "            pb1 =!codonw {chemin}/EssentialsequenceNT-clean.fasta results/{organism[i]}/EssentialcodonwFeature.out results/{organism[i]}/EssentialcodonwFeature.blk -nomenu -nowarn -all_indices -human\n",
    "            print(pb1)\n",
    "            print(\"************* Debut extraction des features de l'organisme\"+\" \"+organism[i] +\"**********\")\n",
    "            #extract sequence feature\n",
    "            !Rscript /content/extract_feature_and_sequences/gepoFeatureEngineering/scriptFeatureBySeq.R {path_out+organism[i]}/EssentialsequenceNT-clean.fasta {path_out+organism[i]}/EssentialsequenceAA-clean.fasta  /content/results/{organism[i]}/Essential\n",
    "            !rm /content/results/{organism[i]}/EssentialcodonwFeature.out\n",
    "            !rm /content/results/{organism[i]}/EssentialcodonwFeature.blk\n",
    "\n",
    "            #iFeature\n",
    "            mkdir /content/results_iFeat_tmp\n",
    "            main_iFeature('/content/sequences_clean/'+organism[i]+'/EssentialsequenceNT-clean.fasta','/content/results_iFeat_tmp/','/content/sequences_clean/'+organism[i]+'/EssentialsequenceAA-clean.fasta','/content/results_iFeat_tmp/')\n",
    "            split_data('/content/results_iFeat_tmp/', '/content/results/'+organism[i]+'/')\n",
    "            !sudo rm -r /content/results_iFeat_tmp\n",
    "\n",
    "            print(\"************* Fin extraction des features de l'organisme\"+\" \"+organism[i] +\"**********\")\n",
    "        else:\n",
    "            print(\"************* Debut extraction des features de l'organisme\"+\" \"+organism[i] +\"**********\")\n",
    "            #extract sequence feature\n",
    "            !Rscript /content/extract_feature_and_sequences/gepoFeatureEngineering/scriptFeatureBySeq.R {path_out+organism[i]}/EssentialsequenceNT-clean.fasta {path_out+organism[i]}/EssentialsequenceAA-clean.fasta /content/results/{organism[i]}/Essential\n",
    "\n",
    "            #iFeature\n",
    "            mkdir /content/results_iFeat_tmp\n",
    "            main_iFeature('/content/sequences_clean/'+organism[i]+'/EssentialsequenceNT-clean.fasta','/content/results_iFeat_tmp/','/content/sequences_clean/'+organism[i]+'/EssentialsequenceAA-clean.fasta','/content/results_iFeat_tmp/')\n",
    "            split_data('/content/results_iFeat_tmp/', '/content/results/'+organism[i]+'/')\n",
    "            !sudo rm -r /content/results_iFeat_tmp\n",
    "\n",
    "            !rm /content/results/{organism[i]}/EssentialcodonwFeature.out\n",
    "            !rm /content/results/{organism[i]}/EssentialcodonwFeature.blk\n",
    "            print(\"************* Fin extraction des features de l'organisme\"+\" \"+organism[i] +\"**********\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "154182db"
   },
   "source": [
    "# execution of the main function for the generation of the organism's features\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "b4abfb29",
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "main(\"/content/extract_feature_and_sequences/organism/\", \"/content/sequences_clean/\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "T6x7VskYoorW"
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "f5d29d81"
   },
   "source": [
    "# Dataset labeling"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "aZ-Q-Xb6nbhi"
   },
   "outputs": [],
   "source": [
    "def main_labeling(epath_file, data_csv):\n",
    "    df = pd.read_csv(data_csv)\n",
    "    # Labelissage du dataset\n",
    "    df['essentiality'] = -1\n",
    "    id_kegg = df.index\n",
    "    epath = pd.read_excel(epath_file, sheet_name='Sheet1')\n",
    "    epath['Gene_Locus'] = epath['Gene_Locus'].str.upper()\n",
    "    epath.set_index(\"Gene_Locus\", inplace=True)\n",
    "    gene_locus = epath.index\n",
    "    for i in range (len(id_kegg)):\n",
    "        if(id_kegg[i] in np.array(gene_locus)):\n",
    "                df.at[id_kegg[i],'essentiality'] = epath.at[id_kegg[i],'Gene_essentaility']\n",
    "    df.drop(df[df.essentiality == -1].index, inplace=True)\n",
    "    mappings = {'NE': 0, 'E': 1}\n",
    "    classes = df.pop('essentiality')\n",
    "    essential_labels = classes.map(mappings)\n",
    "    df['essentiality_status'] = essential_labels\n",
    "    df = df.reset_index(drop=True)\n",
    "    features = df.columns[:-1]\n",
    "    target = [\"essentiality_status\"]\n",
    "    df_pr = preprocessing.normalize(df[features])\n",
    "    df_d = pd.DataFrame(df_pr, columns = features)\n",
    "    target_f = pd.DataFrame(df[\"essentiality_status\"], columns = target)\n",
    "    dataset_norm = pd.concat([df_d, target_f], axis=1)\n",
    "    return dataset_norm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "722e4671"
   },
   "outputs": [],
   "source": [
    "data_epath = find_dir(\"/content/extract_feature_and_sequences/Data_epath/\")\n",
    "for i in range (len(organism_abbr)):\n",
    "  !mkdir /content/{organism_abbr[i]}\n",
    "  for j in range (len(data_epath)):\n",
    "    if(organism_abbr[i] == data_epath[j].split(\"_\")[0]):\n",
    "      organism_data = find_dir(\"/content/results/\"+organism_abbr[i])\n",
    "      for item in range (len(organism_data)):\n",
    "        dataset = main_labeling(\"/content/extract_feature_and_sequences/Data_epath/\"+data_epath[j],\n",
    "                                \"/content/results/\"+organism_abbr[i]+\"/\"+organism_data[item])\n",
    "        subGroupName = organism_data[item].split(\"_\")[2].split(\".\")[0]\n",
    "        dataset.to_csv(organism_abbr[i]+\"/\"+subGroupName+\".csv\")\n",
    "        files.download(subGroupName+\".csv\")\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "6e08f953",
    "outputId": "bdd97954-8787-4a30-dd01-5555eb9af04b"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘dataLabels’: File exists\n",
      "fatal: destination path 'data' already exists and is not an empty directory.\n"
     ]
    }
   ],
   "source": [
    "!cd /content\n",
    "!mkdir dataLabels\n",
    "# !mkdir\n",
    "# !cd data\n",
    "!git clone https://gitlab.com/Berthoazeye/data.git\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "gHltpR9zO_qi",
    "outputId": "c4d46e9d-aeb7-4b80-855d-dfc89551cfb0"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4166"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "input_path = 'data/'\n",
    "output_path = 'dataLabels/'\n",
    "headers = ['Org','Gene_Locus','KO_Nb', 'Gene_Name', 'Gene_Function', 'E_Score','P_Score','Organism']\n",
    "\n",
    "len(os.listdir(input_path))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "id": "xTPZ9lLFUOkm"
   },
   "outputs": [],
   "source": [
    "def supprimer_extension(nom_fichier):\n",
    "    nom_base = os.path.basename(nom_fichier)\n",
    "    base, _ = os.path.splitext(nom_base)\n",
    "    return base"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 1000
    },
    "id": "o5DXGCKMPjWB",
    "outputId": "11f07d5f-4bac-418e-dc29-b080e9c500ff"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "reste---------------------------------------------------------4166\n",
      "data/psn_Pseudopedobacter_saltans.csv\n",
      "reste---------------------------------------------------------4165\n",
      "data/duc_Dehalococcoides_sp._UCH007.csv\n",
      "reste---------------------------------------------------------4164\n",
      "data/mrd_Methylobacterium_radiotolerans.csv\n",
      "reste---------------------------------------------------------4163\n",
      "data/msb_Mycobacterium_smegmatis_MC2_155.csv\n",
      "reste---------------------------------------------------------4162\n",
      "data/bup_Buchnera_aphidicola_TLW03_Acyrthosiphon_pisum).csv\n",
      "/usr/bin/sh: 1: Syntax error: \")\" unexpected\n",
      "reste---------------------------------------------------------4162\n",
      "data/hpx_Helicobacter_pylori_83.csv\n",
      "reste---------------------------------------------------------4161\n",
      "data/ahd_Aeromonas_hydrophila_YL17.csv\n",
      "reste---------------------------------------------------------4160\n",
      "data/spht_Sphingobium_sp._TKS.csv\n",
      "reste---------------------------------------------------------4159\n",
      "data/zmn_Zymomonas_mobilis_subsp._mobilis_NCIMB_11163.csv\n",
      "reste---------------------------------------------------------4158\n",
      "data/mai_Micavibrio_aeruginosavorus_ARL-13.csv\n",
      "reste---------------------------------------------------------4157\n",
      "data/psi_Providencia_stuartii_MRSN_2154.csv\n",
      "reste---------------------------------------------------------4156\n",
      "data/btra_Bibersteinia_trehalosi_USDA-ARS-USMARC-190.csv\n",
      "reste---------------------------------------------------------4155\n",
      "data/acid_Acidovorax_sp_NA2.csv\n",
      "reste---------------------------------------------------------4154\n",
      "data/mef_Methanosarcina_sp._WH1.csv\n",
      "reste---------------------------------------------------------4153\n",
      "data/mthr_Methanosarcina_thermophila_TM-1.csv\n",
      "reste---------------------------------------------------------4152\n",
      "data/lmm_Leuconostoc_mesenteroides_subsp._mesenteroides_J18.csv\n",
      "reste---------------------------------------------------------4151\n",
      "data/seec_Salmonella_enterica_subsp._enterica_Serovar_Cubana.csv\n",
      "reste---------------------------------------------------------4150\n",
      "data/xfa_Xylella_fastidiosa_9a5c.csv\n",
      "reste---------------------------------------------------------4149\n",
      "data/pspw_Paraburkholderia_sprentiae.csv\n",
      "reste---------------------------------------------------------4148\n",
      "data/spl_Shewanella_pealeana.csv\n",
      "reste---------------------------------------------------------4147\n",
      "data/vsp_Vibrio_tasmaniensis.csv\n",
      "reste---------------------------------------------------------4146\n",
      "data/das_Pseudodesulfovibrio_aespoeensis.csv\n",
      "reste---------------------------------------------------------4145\n",
      "data/ptl_Parageobacillus_thermoglucosidans.csv\n",
      "reste---------------------------------------------------------4144\n",
      "data/ocg_Oligotropha_carboxidovorans_OM5_(Goettingen).csv\n",
      "/usr/bin/sh: 1: Syntax error: \"(\" unexpected\n",
      "reste---------------------------------------------------------4144\n",
      "data/axx_Achromobacter_xylosoxidans_NCTC10807.csv\n",
      "reste---------------------------------------------------------4143\n",
      "data/bau_Buchnera_aphidicola_Tuc7_(Acyrthosiphon_pisum).csv\n",
      "/usr/bin/sh: 1: Syntax error: \"(\" unexpected\n",
      "reste---------------------------------------------------------4143\n",
      "data/sii_Sulfolobus_islandicus_L.D.8.5.csv\n",
      "reste---------------------------------------------------------4142\n",
      "data/rpa_Rhodopseudomonas_palustris_CGA009.csv\n",
      "reste---------------------------------------------------------4141\n",
      "data/pcq_Pseudomonas_citronellolis.csv\n",
      "reste---------------------------------------------------------4140\n",
      "data/ttr_Thermobaculum_terrenum.csv\n",
      "reste---------------------------------------------------------4139\n",
      "data/deh_Dehalococcoides_mccartyi_CBDB1.csv\n",
      "reste---------------------------------------------------------4138\n",
      "data/lis_Leptospira_interrogans_serovar_Linhai.csv\n",
      "reste---------------------------------------------------------4137\n",
      "data/ddo_Dokdonia_donghaensis.csv\n",
      "reste---------------------------------------------------------4136\n",
      "data/koy_Klebsiella_michiganensis_HKOPL1.csv\n",
      "reste---------------------------------------------------------4135\n",
      "data/bad_Bifidobacterium_adolescentis_ATCC_15703.csv\n",
      "reste---------------------------------------------------------4134\n",
      "data/cpf_Clostridium_perfringens_ATCC_13124.csv\n",
      "reste---------------------------------------------------------4133\n",
      "data/tpyo_Trueperella_pyogenes_TP8.csv\n",
      "reste---------------------------------------------------------4132\n",
      "data/bpd_Burkholderia_pseudomallei_668.csv\n",
      "reste---------------------------------------------------------4131\n",
      "data/syf_Synechococcus_elongatus_PCC7942.csv\n",
      "reste---------------------------------------------------------4130\n",
      "data/npn_Novosphingobium_pentaromativorans.csv\n",
      "reste---------------------------------------------------------4129\n",
      "data/cama_Citrobacter_amalonaticus.csv\n",
      "reste---------------------------------------------------------4128\n",
      "data/mls_Methanosarcina_lacustris.csv\n",
      "reste---------------------------------------------------------4127\n",
      "data/bli_Bacillus_licheniformis_ATCC_14580 (2).csv\n",
      "/usr/bin/sh: 1: Syntax error: \"(\" unexpected\n",
      "reste---------------------------------------------------------4127\n",
      "data/cgu_Corynebacterium_glutamicum_K051.csv\n",
      "reste---------------------------------------------------------4126\n",
      "data/hyl_Hydrogenophaga_sp._LPB0072.csv\n",
      "reste---------------------------------------------------------4125\n",
      "data/vfl_Vibrio_fluvialis.csv\n",
      "reste---------------------------------------------------------4124\n",
      "data/paci_Pediococcus_acidilactici.csv\n",
      "reste---------------------------------------------------------4123\n",
      "data/lfr_Lactobacillus_fermentum_CECT_5716.csv\n",
      "reste---------------------------------------------------------4122\n",
      "data/mth_Methanothermobacter_thermautotrophicus.csv\n",
      "reste---------------------------------------------------------4121\n",
      "data/cya_Synechococcus_sp._JA-3-3Ab.csv\n",
      "reste---------------------------------------------------------4120\n",
      "data/pma_Prochlorococcus_marinus_SS120.csv\n",
      "reste---------------------------------------------------------4119\n",
      "data/lih_Limnohabitans_sp._63ED37-2.csv\n",
      "reste---------------------------------------------------------4118\n",
      "data/babr_Brucella_abortus_bv_6_870.csv\n",
      "reste---------------------------------------------------------4117\n",
      "data/sho_Streptomyces_hygroscopicus_subsp._jinggangensis_TL01.csv\n",
      "reste---------------------------------------------------------4116\n",
      "data/gbr_Gordonia_bronchialis.csv\n",
      "reste---------------------------------------------------------4115\n",
      "data/aru_Arthrobacter_sp_U41.csv\n",
      "reste---------------------------------------------------------4114\n",
      "data/ial_Ignavibacterium_album.csv\n",
      "reste---------------------------------------------------------4113\n",
      "data/ppj_Pseudomonas_plecoglossicida.csv\n",
      "reste---------------------------------------------------------4112\n",
      "data/sah_Staphylococcus_aureus_subsp._aureus_JH1_(MRSAVSSA).csv\n",
      "/usr/bin/sh: 1: Syntax error: \"(\" unexpected\n",
      "reste---------------------------------------------------------4112\n",
      "data/fbl_Ferrimonas_balearica.csv\n",
      "reste---------------------------------------------------------4111\n",
      "data/jeu_Jeongeupia_sp._USM3.csv\n",
      "reste---------------------------------------------------------4110\n",
      "data/bacp_Bacillus_sp_Pc3.csv\n",
      "reste---------------------------------------------------------4109\n",
      "data/tph_Treponema_pallidum_subsp._pallidum_Chicago.csv\n",
      "reste---------------------------------------------------------4108\n",
      "data/dku_Desulfotomaculum_kuznetsovii.csv\n",
      "reste---------------------------------------------------------4107\n",
      "data/blp_Blattabacterium_sp_(Panesthia_angustipennis_spadica).csv\n",
      "/usr/bin/sh: 1: Syntax error: \"(\" unexpected\n",
      "reste---------------------------------------------------------4107\n",
      "data/fac_Ferroplasma_acidarmanus.csv\n",
      "reste---------------------------------------------------------4106\n",
      "data/ccjz_Corynebacterium_crudilactis.csv\n",
      "reste---------------------------------------------------------4105\n",
      "data/lat_Candidatus_Liberibacter_asiaticus_Ishi-1.csv\n",
      "reste---------------------------------------------------------4104\n",
      "data/mzh_Methanosalsum_zhilinae.csv\n",
      "reste---------------------------------------------------------4103\n",
      "data/cry_Cryobacterium_sp_LW097.csv\n",
      "reste---------------------------------------------------------4102\n",
      "data/ftn_Francisella_tularensis_subsp_novicida_U112.csv\n",
      "reste---------------------------------------------------------4101\n",
      "data/pfn_Pseudomonas_fluorescens_UK4.csv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "reste---------------------------------------------------------4100\n",
      "data/gba_Gemmatirosa_kalamazoonesis.csv\n",
      "reste---------------------------------------------------------4099\n",
      "data/pck_Candidatus_Pantoea_carbekii_US.csv\n",
      "reste---------------------------------------------------------4098\n",
      "data/rsc_Ralstonia_solanacearum_CFBP2957.csv\n",
      "reste---------------------------------------------------------4097\n",
      "data/lam_Lactobacillus_amylovorus_GRL_1112.csv\n",
      "reste---------------------------------------------------------4096\n",
      "data/dmc_Dehalococcoides_mccartyi_BTF08.csv\n",
      "reste---------------------------------------------------------4095\n",
      "data/bang_Bifidobacterium_angulatum.csv\n",
      "reste---------------------------------------------------------4094\n",
      "data/tgr_Thioalkalivibrio_sulfidiphilus.csv\n",
      "reste---------------------------------------------------------4093\n",
      "data/csw__Chlamydia_trachomatis_Sweden2.csv\n",
      "reste---------------------------------------------------------4092\n",
      "data/tcl_Thauera_chlorobenzoica.csv\n",
      "reste---------------------------------------------------------4091\n",
      "data/par_Psychrobacter_arcticus.csv\n",
      "reste---------------------------------------------------------4090\n",
      "data/mck_Mycoplasma_canis_PG_14.csv\n",
      "reste---------------------------------------------------------4089\n",
      "data/mgnc_Mycoplasma_gallisepticum_NC96_1596-4-2P.csv\n",
      "reste---------------------------------------------------------4088\n",
      "data/ctll_Chlamydia_trachomatis_L1440LN.csv\n",
      "reste---------------------------------------------------------4087\n",
      "data/nde_Nitrospira_defluvii.csv\n",
      "reste---------------------------------------------------------4086\n",
      "data/rsv_Rickettsia_slovaca_13-B.csv\n",
      "reste---------------------------------------------------------4085\n",
      "data/cyb_Synechococcus_sp._JA-2-3B'a(2-13).csv\n",
      "/usr/bin/sh: 1: Syntax error: Unterminated quoted string\n",
      "reste---------------------------------------------------------4085\n",
      "data/sln_Staphylococcus_lugdunensis_N920143.csv\n",
      "reste---------------------------------------------------------4084\n",
      "data/mhr_Mycoplasma_hyorhinis_HUB-1.csv\n",
      "reste---------------------------------------------------------4083\n",
      "data/rec_Rhizobium_etli_CIAT_652.csv\n",
      "reste---------------------------------------------------------4082\n",
      "data/sdl_Sulfurospirillum_deleyianum.csv\n",
      "reste---------------------------------------------------------4081\n",
      "data/bcet_Brucella_ceti_TE10759-12.csv\n",
      "reste---------------------------------------------------------4080\n",
      "data/bstl_Burkholderia_stabilis.csv\n",
      "reste---------------------------------------------------------4079\n",
      "data/slg_Staphylococcus_lugdunensis_HKU09-01.csv\n",
      "reste---------------------------------------------------------4078\n",
      "data/lmox_Listeria_monocytogenes_WSLC1042.csv\n",
      "reste---------------------------------------------------------4077\n",
      "data/ctli_Chlamydia_trachomatis_L2434Bu(i).csv\n",
      "/usr/bin/sh: 1: Syntax error: \"(\" unexpected\n",
      "reste---------------------------------------------------------4077\n",
      "data/vcm_Vibrio_cholerae_M66-2.csv\n",
      "reste---------------------------------------------------------4076\n",
      "data/cao_Cellulophaga_algicola.csv\n",
      "reste---------------------------------------------------------4075\n",
      "data/fpw_Flavobacterium_psychrophilum_V4-24.csv\n",
      "reste---------------------------------------------------------4074\n",
      "data/psl_Pirellula_staleyi.csv\n",
      "reste---------------------------------------------------------4073\n",
      "data/ols_Olsenella_uli.csv\n",
      "reste---------------------------------------------------------4072\n",
      "data/wri_Wolbachia_wRi_(Drosophila_simulans).csv\n",
      "/usr/bin/sh: 1: Syntax error: \"(\" unexpected\n",
      "reste---------------------------------------------------------4072\n",
      "data/rle_Rhizobium_leguminosarum_bv._viciae_3841.csv\n",
      "reste---------------------------------------------------------4071\n",
      "data/mpv_Mycoplasma_parvum.csv\n",
      "reste---------------------------------------------------------4070\n",
      "data/upa_Ureaplasma_parvum_serovar_3_ATCC_27815.csv\n",
      "reste---------------------------------------------------------4069\n",
      "data/mtq_Mycobacterium_tuberculosis_HKBS1.csv\n",
      "reste---------------------------------------------------------4068\n",
      "data/cep_Crinalium_epipsammum.csv\n",
      "reste---------------------------------------------------------4067\n",
      "data/papi_Pandoraea_apista.csv\n",
      "reste---------------------------------------------------------4066\n",
      "data/mto_Mycobacterium_tuberculosis_CTRI-2.csv\n",
      "reste---------------------------------------------------------4065\n",
      "data/bsr_Bacillus_subtilis_subsp_subtilis_RO-NN-1.csv\n",
      "reste---------------------------------------------------------4064\n",
      "data/sags_Streptococcus_agalactiae_SA20-06.csv\n",
      "reste---------------------------------------------------------4063\n",
      "data/cab_Chlamydia_abortus_S263.csv\n",
      "reste---------------------------------------------------------4062\n",
      "data/mam_Mesorhizobium_australicum.csv\n",
      "reste---------------------------------------------------------4061\n",
      "data/ctrs_Chlamydia_trachomatis_ESotonE8.csv\n",
      "reste---------------------------------------------------------4060\n",
      "data/tgy_Thermococcus_guaymasensis.csv\n",
      "reste---------------------------------------------------------4059\n",
      "data/esi_Exiguobacterium_sibiricum.csv\n",
      "reste---------------------------------------------------------4058\n",
      "data/mavu_Mycobacterium_avium_subsp._paratuberculosis_E93.csv\n",
      "reste---------------------------------------------------------4057\n",
      "data/efn_Enterococcus_faecalis_DENG1.csv\n",
      "reste---------------------------------------------------------4056\n",
      "data/asr_Aeromonas_schubertii.csv\n",
      "reste---------------------------------------------------------4055\n",
      "data/saur_Staphylococcus_aureus_Bmb9393.csv\n",
      "reste---------------------------------------------------------4054\n",
      "data/srl_Serratia_plymuthica_4Rx13.csv\n",
      "reste---------------------------------------------------------4053\n",
      "data/lcb_Lactobacillus_casei_BL23.csv\n",
      "reste---------------------------------------------------------4052\n",
      "data/tha_Thermococcus_sp._AM4.csv\n",
      "reste---------------------------------------------------------4051\n",
      "data/gfe_Geosporobacter_ferrireducens.csv\n",
      "reste---------------------------------------------------------4050\n",
      "data/bic_Bradyrhizobium_icense.csv\n",
      "reste---------------------------------------------------------4049\n",
      "data/ske_Sanguibacter_keddieii.csv\n",
      "reste---------------------------------------------------------4048\n",
      "data/dth_Dictyoglomus_thermophilum.csv\n",
      "reste---------------------------------------------------------4047\n",
      "data/blr_Brevibacillus_laterosporus.csv\n",
      "reste---------------------------------------------------------4046\n",
      "data/ipo_Ilyobacter_polytropus.csv\n",
      "reste---------------------------------------------------------4045\n",
      "data/bai_Bacillus_anthracis_A0248.csv\n",
      "reste---------------------------------------------------------4044\n",
      "data/sths_Streptococcus_thermophilus_S9.csv\n",
      "reste---------------------------------------------------------4043\n",
      "data/rum_Ruminococcus_sp._SR15.csv\n",
      "reste---------------------------------------------------------4042\n",
      "data/pseo_Pseudoalteromonas_sp._OCN003.csv\n",
      "reste---------------------------------------------------------4041\n",
      "data/lcp_Lysobacter_capsici.csv\n",
      "reste---------------------------------------------------------4040\n",
      "data/vga_Vibrio_gazogenes.csv\n",
      "reste---------------------------------------------------------4039\n",
      "data/eoh_Escherichia_coli_O103H2_12009_(EHEC).csv\n",
      "/usr/bin/sh: 1: Syntax error: \"(\" unexpected\n",
      "reste---------------------------------------------------------4039\n",
      "data/chp_Chlamydia_psittaci_6BC.csv\n",
      "reste---------------------------------------------------------4038\n",
      "data/bfx_Bacillus_flexus.csv\n",
      "reste---------------------------------------------------------4037\n",
      "data/abad_Acinetobacter_baumannii_D1279779.csv\n",
      "reste---------------------------------------------------------4036\n",
      "data/lmk_Leuconostoc_mesenteroides_KFRI-MG.csv\n",
      "reste---------------------------------------------------------4035\n",
      "data/det_Dehalococcoides_mccartyi_195.csv\n",
      "reste---------------------------------------------------------4034\n",
      "data/bsem_Burkholderia_seminalis.csv\n",
      "reste---------------------------------------------------------4033\n",
      "data/bcao_Bacillus_coagulans_DSM_1_=_ATCC_7050.csv\n",
      "reste---------------------------------------------------------4032\n",
      "data/ben_Blochmannia_endosymbiont_of_Camponotus_(Colobopsis)_obliquus_757.csv\n",
      "/usr/bin/sh: 1: Syntax error: \"(\" unexpected\n",
      "reste---------------------------------------------------------4032\n",
      "data/ipa_Isosphaera_pallida.csv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "reste---------------------------------------------------------4031\n",
      "data/tnu_Thermococcus_nautili.csv\n",
      "reste---------------------------------------------------------4030\n",
      "data/bud_Burkholderia_sp_Bp5365.csv\n",
      "reste---------------------------------------------------------4029\n",
      "data/tao_Thioalkalimicrobium_aerophilum.csv\n",
      "reste---------------------------------------------------------4028\n",
      "data/dsh_Dinoroseobacter_shibae.csv\n",
      "reste---------------------------------------------------------4027\n",
      "data/thf_Thermofilum_uzonense.csv\n",
      "reste---------------------------------------------------------4026\n",
      "data/hpyj_Helicobacter_pylori_oki898.csv\n",
      "reste---------------------------------------------------------4025\n",
      "data/nfa_Nocardia_farcinica_IFM_10152.csv\n",
      "reste---------------------------------------------------------4024\n",
      "data/edh_Escherichia_coli_DH1.csv\n",
      "reste---------------------------------------------------------4023\n",
      "data/ftr_Francisella_tularensis_subsp_tularensis_NE061598.csv\n",
      "reste---------------------------------------------------------4022\n",
      "data/bamp_Bacillus_velezensis_AS43.3.csv\n",
      "reste---------------------------------------------------------4021\n",
      "data/asa_Aeromonas_salmonicida.csv\n",
      "reste---------------------------------------------------------4020\n",
      "data/bme_Brucella_melitensis_bv_1_16M.csv\n",
      "reste---------------------------------------------------------4019\n",
      "data/pvi_Chlorobiumu_phaeovibrioides.csv\n",
      "reste---------------------------------------------------------4018\n",
      "data/cje_Campylobacter_jejuni_subsp_jejuni_NCTC_11168_=_ATCC_700819.csv\n",
      "reste---------------------------------------------------------4017\n",
      "data/paei_Pseudomonas_aeruginosa_PAO1-VE2.csv\n",
      "reste---------------------------------------------------------4016\n",
      "data/eih_Escherichia_coli_IHE3034.csv\n",
      "reste---------------------------------------------------------4015\n",
      "data/mtuc_Mycobacterium_tuberculosis_CAS_NITR204.csv\n",
      "reste---------------------------------------------------------4014\n",
      "data/cthj_Chlamydia_trachomatis_RC-J953.csv\n",
      "reste---------------------------------------------------------4013\n",
      "data/shl_Shewanella_halifaxensis.csv\n",
      "reste---------------------------------------------------------4012\n",
      "data/fnu_Fusobacterium_nucleatum_subsp_nucleatum_ATCC_25586.csv\n",
      "reste---------------------------------------------------------4011\n",
      "data/hlr_Halostagnicola_larsenii.csv\n",
      "reste---------------------------------------------------------4010\n",
      "data/sds_Streptococcus_dysgalactiae_subsp._equisimilis_GGS.csv\n",
      "reste---------------------------------------------------------4009\n",
      "data/lpu_Legionella_pneumophila_subsp._pneumophila_LPE509.csv\n",
      "reste---------------------------------------------------------4008\n",
      "data/amt_Alkaliphilus_metalliredigens.csv\n",
      "reste---------------------------------------------------------4007\n",
      "data/xac_Xanthomonas_citri_pv._citri_306.csv\n",
      "reste---------------------------------------------------------4006\n",
      "data/cps_Colwellia_ psychrerythraea.csv\n",
      "rm: cannot remove 'data/cps_Colwellia_': No such file or directory\n",
      "rm: cannot remove 'psychrerythraea.csv': No such file or directory\n",
      "reste---------------------------------------------------------4006\n",
      "data/ctrq_Chlamydia_trachomatis_A363.csv\n",
      "reste---------------------------------------------------------4005\n",
      "data/pep_Pedobacter_sp._PACM_27299.csv\n",
      "reste---------------------------------------------------------4004\n",
      "data/stc_Streptococcus_thermophilus_CNRZ1066.csv\n",
      "reste---------------------------------------------------------4003\n",
      "data/sfh_Sinorhizobium_fredii_HH103.csv\n",
      "reste---------------------------------------------------------4002\n",
      "data/bpse_Burkholderia pseudomallei MSHR305.csv\n",
      "rm: cannot remove 'data/bpse_Burkholderia': No such file or directory\n",
      "rm: cannot remove 'pseudomallei': No such file or directory\n",
      "rm: cannot remove 'MSHR305.csv': No such file or directory\n",
      "reste---------------------------------------------------------4002\n",
      "data/hxa_Halopiger_xanaduensis.csv\n",
      "reste---------------------------------------------------------4001\n",
      "data/hil_Haemophilus_influenzae_F3047_(nontypeable).csv\n",
      "/usr/bin/sh: 1: Syntax error: \"(\" unexpected\n",
      "reste---------------------------------------------------------4001\n",
      "data/ahi_Aeromonas_hydrophila_AH10.csv\n",
      "reste---------------------------------------------------------4000\n",
      "data/cmx_Chlamydia_muridarum_Nigg3.csv\n",
      "reste---------------------------------------------------------3999\n",
      "data/pstw_Pantoea_stewartii_subsp._stewartii.csv\n",
      "reste---------------------------------------------------------3998\n",
      "data/cfz_Campylobacter_fetus_subsp_venerealis_84-112.csv\n",
      "reste---------------------------------------------------------3997\n",
      "data/ppol_Paenibacillus_polymyxa_CR1.csv\n",
      "reste---------------------------------------------------------3996\n",
      "data/sij_Scardovia_inopinata.csv\n",
      "reste---------------------------------------------------------3995\n",
      "data/san_Streptococcus_agalactiae_NEM316_(serotype_III).csv\n",
      "/usr/bin/sh: 1: Syntax error: \"(\" unexpected\n",
      "reste---------------------------------------------------------3995\n",
      "data/mgw_Mycoplasma_gallisepticum_WI01_2001.043-13-2P.csv\n",
      "reste---------------------------------------------------------3994\n",
      "data/bbre_Bifidobacterium_breve_12L.csv\n",
      "reste---------------------------------------------------------3993\n",
      "data/abt_Arcobacter_butzleri_ED-1.csv\n",
      "reste---------------------------------------------------------3992\n",
      "data/phe_Pedobacter_heparinus.csv\n",
      "reste---------------------------------------------------------3991\n",
      "data/hed_Candidatus_Hoaglandella_endobia.csv\n",
      "reste---------------------------------------------------------3990\n",
      "data/bmp_Bacillus_velezensis.csv\n",
      "reste---------------------------------------------------------3989\n",
      "data/pcg_Pseudomonas_corrugata.csv\n",
      "reste---------------------------------------------------------3988\n",
      "data/ypc_Yersinia_pseudotuberculosis_1.csv\n",
      "reste---------------------------------------------------------3987\n",
      "data/tvo_Thermoplasma_volcanium.csv\n",
      "reste---------------------------------------------------------3986\n",
      "data/sten_Stenotrophomonas_sp._WZN-1.csv\n",
      "reste---------------------------------------------------------3985\n",
      "data/lpr_Lactobacillus_plantarum_subsp._plantarum_P-8.csv\n",
      "reste---------------------------------------------------------3984\n",
      "data/ypr_Yersinia_pseudotuberculosis_EP2+.csv\n",
      "reste---------------------------------------------------------3983\n",
      "data/kpk_Klebsiella_pneumoniae_subsp._pneumoniae_KP5-1.csv\n",
      "reste---------------------------------------------------------3982\n",
      "data/saub_Staphylococcus_aureus_08BA02176.csv\n",
      "reste---------------------------------------------------------3981\n",
      "data/vvl_Vibrio_vulnificus_93U204.csv\n",
      "reste---------------------------------------------------------3980\n",
      "data/pdg_Planococcus_donghaensis.csv\n",
      "reste---------------------------------------------------------3979\n",
      "data/senr_Salmonella_enterica_subsp._enterica_serovar_Typhimurium_DT2.csv\n",
      "reste---------------------------------------------------------3978\n",
      "data/pra_Cutibacterium_avidum.csv\n",
      "reste---------------------------------------------------------3977\n",
      "data/nmh_Neisseria_meningitidis_H44_76_(serogroup_B).csv\n",
      "/usr/bin/sh: 1: Syntax error: \"(\" unexpected\n",
      "reste---------------------------------------------------------3977\n",
      "data/afl_Anoxybacillus_amylolyticus.csv\n",
      "reste---------------------------------------------------------3976\n",
      "data/bmf_Brucella_abortus_2308.csv\n",
      "reste---------------------------------------------------------3975\n",
      "data/ain_Acidaminococcus_intestini.csv\n",
      "reste---------------------------------------------------------3974\n",
      "data/ebw_Escherichia_coli_BW2952.csv\n",
      "reste---------------------------------------------------------3973\n",
      "data/gvg_Gardnerella_vaginalis_ATCC_14019.csv\n",
      "reste---------------------------------------------------------3972\n",
      "data/cur_Corynebacterium_urealyticum_DSM_7109.csv\n",
      "reste---------------------------------------------------------3971\n",
      "data/mcj_Methanosaeta_concilii.csv\n",
      "reste---------------------------------------------------------3970\n",
      "data/osp_Odoribacter_splanchnicus.csv\n",
      "reste---------------------------------------------------------3969\n",
      "data/lac_Lactobacillus_acidophilus_NCFM.csv\n",
      "reste---------------------------------------------------------3968\n",
      "data/lmoq_Listeria_monocytogenes_6179.csv\n",
      "reste---------------------------------------------------------3967\n",
      "data/amd_Amycolatopsis_mediterranei_U32.csv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "reste---------------------------------------------------------3966\n",
      "data/llt_Lactococcus_lactis_subsp._lactis_CV56.csv\n",
      "reste---------------------------------------------------------3965\n",
      "data/tti_Thioalkalivibrio_thiocyanoxidans.csv\n",
      "reste---------------------------------------------------------3964\n",
      "data/mek_Methanosarcina_sp._Kolksee.csv\n",
      "reste---------------------------------------------------------3963\n",
      "data/xpe_Xanthomonas_perforans.csv\n",
      "reste---------------------------------------------------------3962\n",
      "data/mgh_Mycoplasma_gallisepticum_R(high).csv\n",
      "/usr/bin/sh: 1: Syntax error: \"(\" unexpected\n",
      "reste---------------------------------------------------------3962\n",
      "data/ptt_Photorhabdus_temperata.csv\n",
      "reste---------------------------------------------------------3961\n",
      "data/slr_Salinispira_pacifica.csv\n",
      "reste---------------------------------------------------------3960\n",
      "data/smi_Sinorhizobium_meliloti_Rm41.csv\n",
      "reste---------------------------------------------------------3959\n",
      "data/mcac_Mycoplasma_capricolum_subsp._capripneumoniae_9231-Abomsa.csv\n",
      "reste---------------------------------------------------------3958\n",
      "data/psos_Pseudomonas_sp._Os17.csv\n",
      "reste---------------------------------------------------------3957\n",
      "data/bcv_Beutenbergia_cavernae.csv\n",
      "reste---------------------------------------------------------3956\n",
      "data/lcu_Lactobacillus_curieae.csv\n",
      "reste---------------------------------------------------------3955\n",
      "data/asi_Actinobacillus_suis_H91-0380.csv\n",
      "reste---------------------------------------------------------3954\n",
      "data/mcy_Mycoplasma_cynos.csv\n",
      "reste---------------------------------------------------------3953\n",
      "data/hpas_Haemophilus_parasuis_SH03.csv\n",
      "reste---------------------------------------------------------3952\n",
      "data/bxi_Bacillus_xiamenensis.csv\n",
      "reste---------------------------------------------------------3951\n",
      "data/psee_Pseudonocardia_sp._EC080625-04.csv\n",
      "reste---------------------------------------------------------3950\n",
      "data/mej_Methylophaga_nitratireducenticrescens.csv\n",
      "reste---------------------------------------------------------3949\n",
      "data/babc_Brucella_abortus_NCTC_10505.csv\n",
      "reste---------------------------------------------------------3948\n",
      "data/tac_Thermoplasma_acidophilum.csv\n",
      "reste---------------------------------------------------------3947\n",
      "data/sey_Salmonella_enterica_subsp._enterica_serovar_Typhimurium_SL1344.csv\n",
      "reste---------------------------------------------------------3946\n",
      "data/meo_Candidatus_Moranella_endobia_PCVAL.csv\n",
      "reste---------------------------------------------------------3945\n",
      "data/mfm_Mycoplasma_fermentans_M64.csv\n",
      "reste---------------------------------------------------------3944\n",
      "data/eno_Enterobacter_cloacae_subsp_cloacae_ENHKU01.csv\n",
      "reste---------------------------------------------------------3943\n",
      "data/pbu_Phycisphaerae_bacterium_L21-RPul-D3.csv\n",
      "reste---------------------------------------------------------3942\n",
      "data/shp_Shewanella_putrefaciens_200.csv\n",
      "reste---------------------------------------------------------3941\n",
      "data/cgx_Corynebacterium_glutamicum_B253.csv\n",
      "reste---------------------------------------------------------3940\n",
      "data/hbc_Hyphomonadaceae_bacterium_UKL13-1.csv\n",
      "reste---------------------------------------------------------3939\n",
      "data/ssal_Spiribacter_salinus.csv\n",
      "reste---------------------------------------------------------3938\n",
      "data/mge_Mycoplasma_genitalium_G37.csv\n",
      "reste---------------------------------------------------------3937\n",
      "data/mari_Marinobacter_sp._CP1.csv\n",
      "reste---------------------------------------------------------3936\n",
      "data/rpn_Rickettsia_prowazekii_NMRC_Madrid_E.csv\n",
      "reste---------------------------------------------------------3935\n",
      "data/pdn_Prevotella_denticola.csv\n",
      "reste---------------------------------------------------------3934\n",
      "data/thv_Thermococcus_sp._2319x1.csv\n",
      "reste---------------------------------------------------------3933\n",
      "data/hph_Helicobacter_pylori_Lithuania75.csv\n",
      "reste---------------------------------------------------------3932\n",
      "data/hma_Haloarcula_marismortui.csv\n",
      "reste---------------------------------------------------------3931\n",
      "data/hty_Helicobacter_typhlonius.csv\n",
      "reste---------------------------------------------------------3930\n",
      "data/pes_Candidatus_Sodalis_pierantonius.csv\n",
      "reste---------------------------------------------------------3929\n",
      "data/ypf_Yersinia_pseudotuberculosis_PA3606.csv\n",
      "reste---------------------------------------------------------3928\n",
      "data/krh_Kocuria_rhizophila.csv\n",
      "reste---------------------------------------------------------3927\n",
      "data/fth_Francisella_tularensis_subsp_holarctica_OSU18.csv\n",
      "reste---------------------------------------------------------3926\n",
      "data/kpw_Klebsiella_pneumoniae_subsp._pneumoniae_KPNIH30.csv\n",
      "reste---------------------------------------------------------3925\n",
      "data/sav_Staphylococcus_aureus_subsp._aureus_Mu50_(MRSAVISA).csv\n",
      "/usr/bin/sh: 1: Syntax error: \"(\" unexpected\n",
      "reste---------------------------------------------------------3925\n",
      "data/baf_Borrelia_afzelii_(FLI).csv\n",
      "/usr/bin/sh: 1: Syntax error: \"(\" unexpected\n",
      "reste---------------------------------------------------------3925\n",
      "data/bmec_Burkholderia_metallica.csv\n",
      "reste---------------------------------------------------------3924\n",
      "data/lru_Lactobacillus_reuteri_SD2112.csv\n",
      "reste---------------------------------------------------------3923\n",
      "data/nme_Neisseria_meningitidis_MC58_(serogroup_B).csv\n",
      "/usr/bin/sh: 1: Syntax error: \"(\" unexpected\n",
      "reste---------------------------------------------------------3923\n",
      "data/hpyi_Helicobacter_pylori_UM037.csv\n",
      "reste---------------------------------------------------------3922\n",
      "data/rpz_Rickettsia_prowazekii_Dachau.csv\n",
      "reste---------------------------------------------------------3921\n",
      "data/avi_Agrobacterium_vitis.csv\n",
      "reste---------------------------------------------------------3920\n",
      "data/bfi_Butyrate-producing_bacterium_SS34.csv\n",
      "reste---------------------------------------------------------3919\n",
      "data/bbau_Bacteroidetes_bacterium.csv\n",
      "reste---------------------------------------------------------3918\n",
      "data/caa_Coraliomargarita_akajimensis.csv\n",
      "reste---------------------------------------------------------3917\n",
      "data/bbe_Brevibacillus_brevis.csv\n",
      "reste---------------------------------------------------------3916\n",
      "data/vha_Vibrio_campbellii.csv\n",
      "reste---------------------------------------------------------3915\n",
      "data/oac_Oscillatoria_acuminata.csv\n",
      "reste---------------------------------------------------------3914\n",
      "data/ann_Aggregatibacter_actinomycetemcomitans_D7S-1.csv\n",
      "reste---------------------------------------------------------3913\n",
      "data/bsn_Bacillus_subtilis_BSn5.csv\n",
      "reste---------------------------------------------------------3912\n",
      "data/hct_Candidatus_Hodgkinia_cicadicola_TETULN.csv\n",
      "reste---------------------------------------------------------3911\n",
      "data/mil_Micromonospora_sp._L5.csv\n",
      "reste---------------------------------------------------------3910\n",
      "data/ecz_Escherichia_coli_O45K1H7_S88_(ExPEC).csv\n",
      "/usr/bin/sh: 1: Syntax error: \"(\" unexpected\n",
      "reste---------------------------------------------------------3910\n",
      "data/nmo_Natronomonas_moolapensis.csv\n",
      "reste---------------------------------------------------------3909\n",
      "data/psy_Psychromonas_sp._CNPT3.csv\n",
      "reste---------------------------------------------------------3908\n",
      "data/ftd_Francisella_tularensis_subsp_novicida_F6168.csv\n",
      "reste---------------------------------------------------------3907\n",
      "data/cbab_omamonadaceae_bacterium_B1.csv\n",
      "reste---------------------------------------------------------3906\n",
      "data/blz_Bifidobacterium_longum_subsp_longum_GT15.csv\n",
      "reste---------------------------------------------------------3905\n",
      "data/pst_Pseudomonas_syringae_pv._tomato_DC3000.csv\n",
      "reste---------------------------------------------------------3904\n",
      "data/cjy_Corynebacterium_glyciniphilum.csv\n",
      "reste---------------------------------------------------------3903\n",
      "data/blu_Blattabacterium_sp_(Nauphoeta_cinerea).csv\n",
      "/usr/bin/sh: 1: Syntax error: \"(\" unexpected\n",
      "reste---------------------------------------------------------3903\n",
      "data/mwe_Mycoplasma_wenyonii.csv\n",
      "reste---------------------------------------------------------3902\n",
      "data/ttn_Thermoproteus_tenax.csv\n",
      "reste---------------------------------------------------------3901\n",
      "data/bxb_Paraburkholderia_xenovorans_LB400.csv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "reste---------------------------------------------------------3900\n",
      "data/slc_Streptomyces_lydicus.csv\n",
      "reste---------------------------------------------------------3899\n",
      "data/acx_Azotobacter_chroococcum.csv\n",
      "reste---------------------------------------------------------3898\n",
      "data/tna_Thermotoga_neapolitana.csv\n",
      "reste---------------------------------------------------------3897\n",
      "data/sphj_Sphingomonas_sp._JJ-A5.csv\n",
      "reste---------------------------------------------------------3896\n",
      "data/vpa_Vibrio_parahaemolyticus_RIMD_2210633.csv\n",
      "reste---------------------------------------------------------3895\n",
      "data/pri_Paenibacillus_riograndensis.csv\n",
      "reste---------------------------------------------------------3894\n",
      "data/meso_Mesorhizobium_sp._B7.csv\n",
      "reste---------------------------------------------------------3893\n",
      "data/cac_Clostridium_acetobutylicum_ATCC_824.csv\n",
      "reste---------------------------------------------------------3892\n",
      "data/mhx_Mannheimia_haemolytica_M42548.csv\n",
      "reste---------------------------------------------------------3891\n",
      "data/pec_Pectobacterium_sp._SCC3193.csv\n",
      "reste---------------------------------------------------------3890\n",
      "data/ppuu_Pseudomonas_sp._UW4.csv\n",
      "reste---------------------------------------------------------3889\n",
      "data/ctch_Chlamydia_trachomatis_F11-96.csv\n",
      "reste---------------------------------------------------------3888\n",
      "data/mcl_Macrococcus_caseolyticus.csv\n",
      "reste---------------------------------------------------------3887\n",
      "data/aps_Candidatus_Azobacteroides_pseudotrichonymphae_(Coptotermes_formosanus).csv\n",
      "/usr/bin/sh: 1: Syntax error: \"(\" unexpected\n",
      "reste---------------------------------------------------------3887\n",
      "data/bchi_Borreliella_chilensis.csv\n",
      "reste---------------------------------------------------------3886\n",
      "data/bti_Bacillus_thuringiensis_HD-771.csv\n",
      "reste---------------------------------------------------------3885\n",
      "data/vcy_Vibrio_coralliilyticus_RE98.csv\n",
      "reste---------------------------------------------------------3884\n",
      "data/asm_Candidatus_Arthromitus_sp_SFB-mouse-Yit.csv\n",
      "reste---------------------------------------------------------3883\n",
      "data/ccoo_Campylobacter_coli_OR12.csv\n",
      "reste---------------------------------------------------------3882\n",
      "data/cfc_Corynebacterium_flavescens.csv\n",
      "reste---------------------------------------------------------3881\n",
      "data/bac_Burkholderia_ambifaria_MC40-6.csv\n",
      "reste---------------------------------------------------------3880\n",
      "data/lwe_Listeria_welshimeri.csv\n",
      "reste---------------------------------------------------------3879\n",
      "data/sgr_Streptomyces_griseus.csv\n",
      "reste---------------------------------------------------------3878\n",
      "data/fre_Frankia_sp_EAN1pec.csv\n",
      "reste---------------------------------------------------------3877\n",
      "data/asp_Alteromonas_stellipolaris_LMG_21856.csv\n",
      "reste---------------------------------------------------------3876\n",
      "data/dts_Delftia_tsuruhatensis.csv\n",
      "reste---------------------------------------------------------3875\n",
      "data/bbay_Burkholderiales_bacterium_YL45.csv\n",
      "reste---------------------------------------------------------3874\n",
      "data/llu_Labilithrix_luteola.csv\n",
      "reste---------------------------------------------------------3873\n",
      "data/lmf_Listeria_monocytogenes_F2365.csv\n",
      "reste---------------------------------------------------------3872\n",
      "data/eab_Escherichia_coli_ABU_83972.csv\n",
      "reste---------------------------------------------------------3871\n",
      "data/hiz_Haemophilus_influenzae_R2866_(nontypeable).csv\n",
      "/usr/bin/sh: 1: Syntax error: \"(\" unexpected\n",
      "reste---------------------------------------------------------3871\n",
      "data/hhd_Halobacillus_halophilus.csv\n",
      "reste---------------------------------------------------------3870\n",
      "data/kpb_Klebsiella_pneumoniae_blaNDM-1.csv\n",
      "reste---------------------------------------------------------3869\n",
      "data/hem_Helicobacter_pylori_UM298.csv\n",
      "reste---------------------------------------------------------3868\n",
      "data/sch_Sphingobium_chlorophenolicum.csv\n",
      "reste---------------------------------------------------------3867\n",
      "data/nso_Niabella_soli.csv\n",
      "reste---------------------------------------------------------3866\n",
      "data/shb_Salmonella_enterica_subsp._enterica_serovar_Heidelberg_B182.csv\n",
      "reste---------------------------------------------------------3865\n",
      "data/btj_Burkholderia_thailandensis_E444.csv\n",
      "reste---------------------------------------------------------3864\n",
      "data/xor_Xanthomonas_oryzae_pv._oryzicola_BLS256.csv\n",
      "reste---------------------------------------------------------3863\n",
      "data/sea_Salmonella_enterica_subsp._enterica_serovar_Agona_SL483.csv\n",
      "reste---------------------------------------------------------3862\n",
      "data/smc_Streptococcus_mutans_NN2025.csv\n",
      "reste---------------------------------------------------------3861\n",
      "data/pyc_Pyrococcus_kukulkanii.csv\n",
      "reste---------------------------------------------------------3860\n",
      "data/ctcj_Chlamydia_trachomatis_RC-J943.csv\n",
      "reste---------------------------------------------------------3859\n",
      "data/mga_Mycoplasma_gallisepticum_R(low).csv\n",
      "/usr/bin/sh: 1: Syntax error: \"(\" unexpected\n",
      "reste---------------------------------------------------------3859\n",
      "data/cav_Chlamydia_avium.csv\n",
      "reste---------------------------------------------------------3858\n",
      "data/ppb_Pseudomonas_putida_BIRD-1.csv\n",
      "reste---------------------------------------------------------3857\n",
      "data/xct_Xanthomonas_citri_subsp._citri_A306.csv\n",
      "reste---------------------------------------------------------3856\n",
      "data/hei_Helicobacter_pylori_Rif2.csv\n",
      "reste---------------------------------------------------------3855\n",
      "data/aas_Candidatus_Amoebophilus_asiaticus.csv\n",
      "reste---------------------------------------------------------3854\n",
      "data/cmi_Clavibacter_michiganensis_subsp_michiganensis.csv\n",
      "reste---------------------------------------------------------3853\n",
      "data/sxy_Staphylococcus_xylosus_HKUOPL8.csv\n",
      "reste---------------------------------------------------------3852\n",
      "data/psul_Pseudarthrobacter_sulfonivorans.csv\n",
      "reste---------------------------------------------------------3851\n",
      "data/tsu_Treponema_succinifaciens.csv\n",
      "reste---------------------------------------------------------3850\n",
      "data/hal_Halobacterium_salinarum_NRC-1.csv\n",
      "reste---------------------------------------------------------3849\n",
      "data/pcm_Pedobacter_cryoconitis.csv\n",
      "reste---------------------------------------------------------3848\n",
      "data/agr_Agrobacterium_sp_H13-3.csv\n"
     ]
    }
   ],
   "source": [
    "for filename in os.listdir(input_path):\n",
    "    print(\"reste---------------------------------------------------------\"+str(len(os.listdir(input_path))))\n",
    "    f = os.path.join(input_path, filename)\n",
    "    print(f)\n",
    "    if os.path.isfile(f):\n",
    "        file_csv = pd.read_csv(f)\n",
    "        for val in range(0,len(file_csv)):\n",
    "            # print(file_csv.loc[val,'E_Score'])\n",
    "            e_score = float(file_csv.loc[val,'E_Score'])\n",
    "            file_csv.replace(to_replace =\"9.16E-\",\n",
    "                 value = \"0.000\",\n",
    "                  inplace = True)\n",
    "            file_csv.replace(to_replace =\"4.28E-\",\n",
    "                 value = \"0.000\",\n",
    "                  inplace = True)\n",
    "            file_csv.replace(to_replace =\"3.47E-\",\n",
    "                 value = \"0.000\",\n",
    "                  inplace = True)\n",
    "            file_csv.replace(to_replace =\"5.20E-\",\n",
    "                 value = \"0.000\",\n",
    "                  inplace = True)\n",
    "            file_csv.replace(to_replace =\"9.33E-\",\n",
    "                 value = \"0.000\",\n",
    "                  inplace = True)\n",
    "            file_csv.replace(to_replace =\"4.31E-\",\n",
    "                 value = \"0.000\",\n",
    "                  inplace = True)\n",
    "            file_csv.replace(to_replace =\"2.64E-\",\n",
    "                 value = \"0.000\",\n",
    "                  inplace = True)\n",
    "            file_csv.replace(to_replace =\"3.23E-\",\n",
    "                 value = \"0.000\",\n",
    "                  inplace = True)\n",
    "            file_csv.replace(to_replace =\"5.82E-\",\n",
    "                 value = \"0.000\",\n",
    "                  inplace = True)\n",
    "            file_csv.replace(to_replace =\"6.11E-\",\n",
    "                 value = \"0.000\",\n",
    "                  inplace = True)\n",
    "            file_csv.replace(to_replace =\"9.59E-\",\n",
    "                 value = \"0.000\",\n",
    "                  inplace = True)\n",
    "            # print(file_csv.loc[val,'P_Score'])\n",
    "            p_score = float(file_csv.loc[val,'P_Score'])\n",
    "            # print(p_score)\n",
    "            if (p_score!=0 and e_score!=0):\n",
    "                if e_score>=0.6:\n",
    "                    file_csv.loc[val, 'Gene_essentaility'] = 'E'\n",
    "                else:\n",
    "                    file_csv.loc[val, 'Gene_essentaility'] = 'NE'\n",
    "            else:\n",
    "                if (p_score>=0.03 or e_score>=0.6):\n",
    "                    file_csv.loc[val, 'Gene_essentaility'] = 'E'\n",
    "                else:\n",
    "                    file_csv.loc[val, 'Gene_essentaility'] = 'NE'\n",
    "    #file_csv.to_csv(f)\n",
    "    # !rm /content/results/{organism[i]}/EssentialcodonwFeature.out\n",
    "    !rm {input_path}{filename}\n",
    "    file_csv.to_excel(output_path+supprimer_extension(filename)+'.xlsx', index= None, header=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "69kjaER7OUgF",
    "outputId": "34fa825d-1135-4b6a-ae3a-6304eb30a02e"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cloning into 'data'...\n",
      "remote: Enumerating objects: 4647, done.\u001b[K\n",
      "remote: Counting objects: 100% (4647/4647), done.\u001b[K\n",
      "remote: Compressing objects: 100% (1486/1486), done.\u001b[K\n",
      "remote: Total 4647 (delta 3436), reused 4367 (delta 3160), pack-reused 0\u001b[K\n",
      "Receiving objects: 100% (4647/4647), 65.83 MiB | 19.77 MiB/s, done.\n",
      "Resolving deltas: 100% (3436/3436), done.\n",
      "Updating files: 100% (4627/4627), done.\n"
     ]
    }
   ],
   "source": [
    "!git clone https://gitlab.com/Berthoazeye/data.git "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "uCxpmzVNOiz8",
    "outputId": "c13d9ca2-f87d-47b0-8b9b-98c08f6688e7"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "error: remote origin1 already exists.\n",
      "Host key verification failed.\n",
      "fatal: Could not read from remote repository.\n",
      "\n",
      "Please make sure you have the correct access rights\n",
      "and the repository exists.\n"
     ]
    }
   ],
   "source": [
    "!cd /content/dataLabels\n",
    "\n",
    "# !git config --global user.email \"romeoberthoazeye@gmail.com\"\n",
    "# !git config --global user.name \"berthoazeyeh\"\n",
    "\n",
    "# !git init\n",
    "# !git add .\n",
    "# !git commit -m \"data\"\n",
    "# !git branch -M main\n",
    "!git remote add origin1 git@github.com:berthoazeyeh/EPATHDATAS.git\n",
    "!git push -u origin1 main"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "tw6dyYDq021y",
    "outputId": "007009d8-6271-49fc-b220-ab99e15a6ed4"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "^C\n",
      "fatal: could not read Username for 'https://github.com': No such device or address\n"
     ]
    }
   ],
   "source": [
    "# !ssh-keygen\n",
    "# !git push --set-upstream origin main\n",
    "\n",
    "# !ls /content/.ssh/id_rsa\n",
    "\n",
    "!git credential-cache exit\n",
    "!git credential-cache store\n",
    "\n",
    "\n",
    "!git branch -M master\n",
    "!git push -u origin master"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "id": "0T0ruIvD1NkK"
   },
   "outputs": [],
   "source": [
    "import zipfile\n",
    "import os\n",
    "\n",
    "def compresser_dossier(source_dir, output_zip):\n",
    "    \"\"\"\n",
    "    Compresser les données d'un dossier en un fichier ZIP.\n",
    "\n",
    "    Args:\n",
    "        source_dir (str): Chemin du dossier source à compresser.\n",
    "        output_zip (str): Chemin du fichier ZIP de sortie.\n",
    "\n",
    "    Returns:\n",
    "        None\n",
    "    \"\"\"\n",
    "    with zipfile.ZipFile(output_zip, 'w', zipfile.ZIP_DEFLATED) as zipf:\n",
    "        for root, dirs, files in os.walk(source_dir):\n",
    "            for file in files:\n",
    "                file_path = os.path.join(root, file)\n",
    "                zipf.write(file_path, os.path.relpath(file_path, source_dir))\n",
    "\n",
    "# Exemple d'utilisation :\n",
    "dossier_source = '/content/data'\n",
    "fichier_zip_sortie = '/content/fichier33.zip'\n",
    "\n",
    "compresser_dossier(dossier_source, fichier_zip_sortie)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "lPD0MFf-5Ax6"
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "celltoolbar": "Format de la Cellule Texte Brut",
  "colab": {
   "provenance": []
  },
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
